### PR TITLE
IonicSwapV1

### PR DESCRIPTION
**Pull request by salmanferrum**
1. Integrated and Updated the FIBER Engine IONIC Swap that swaps Ionic Assets per the documentation [here](https://docs.ferrumnetwork.io/ferrum-network-ecosystem/v/infinityswap-and-bridge-architecture-overview/architecture-and-tech-stack/general-architecture-overview/core-components-of-multiswap/fiber-ferrum-inter-blockchain-express-routing-engine/asset-categorization-and-route-optimization/example-of-asset-categorization)
2. Updated the Functionality FIBER Engine IONIC Asset Swap Check that checks if the assets are IONIC. 
3. Improved the functionality of Swapping IonicAssets / RefineryAssets within FiberRouter to process all Refinery/IONIC assets Swaps
4. Tested the IONIC Swap and performed several successful IONIC swaps from 
- goerliADA [ethTx](https://goerli.etherscan.io/tx/0xfdc0e132a04b3849b0c96720a3bf270d3075f8287455b097d2de54ded094a695) -> BScCAKE [bscTx](https://testnet.bscscan.com/tx/0x1c09f71431823a0aaaf3d37c6d2326e4957163ab2f19006d83e3b6cf7e5cbb07)
- goerliADA [ethTx](https://goerli.etherscan.io/tx/0x2c0ccc5469703f1810f027da7d59424fd9b58cf6f577cc20381e4680f95b7da7)-> BScADA [bscTx](https://testnet.bscscan.com/tx/0x1617e402f45b6163b83f63cdf7d7cfc78c8b4c04679481b1ada76944b1d7e3f3)

 
**Please NOTE:** 

- **Foundry.js** is specific to swapping only Foundry/Bridgeable assets
- **Refinery.js** is specific to swapping Refinery assets
- **Ionic.js** is specific to swapping Ionic assets

### CODE REVIEW

File: contracts/upgradeable-Bridge/FiberRouter.sol [hunk 1]
```diff
@@ -37,7 +33,38 @@ contract FiberRouter is ReentrancyGuardUpgradeable, OwnableUpgradeable {
...
+            ,
+            ,
 
+        ) = /*uint startedAt*/
+            /*uint timeStamp*/
+            /*uint80 answeredInRound*/
+            priceFeed[_token].latestRoundData();
+        uint8 baseDecimals = priceFeed[_token].decimals();
+        return uint256(price) * 10**(18 - baseDecimals);
+        // return uint(price);
```
Comment (jpine3528) [inline contracts/upgradeable-Bridge/FiberRouter.sol:66]: I think dead code could be removed.

File: scripts/Ionic.js [hunk 2]
```diff
@@ -119,33 +127,35 @@ async function targetFACCheck(tokenAddress, amount) {
...
-  if (isTargetTokenFoundryAsset == true && Number(targetFoundryAssetLiquidity) > Number(amount)) {
-      return true;
-  }
-  else{
+  if (
+    isTargetTokenFoundryAsset == true &&
+    Number(targetFoundryAssetLiquidity) > Number(amount)
+  ) {
+    return true;
+  } else {
```
Comment (jpine3528) [inline scripts/Ionic.js:140]: I think `else {` could be removed and `return false` directly for code optimization.

File: scripts/Ionic.js [hunk 3]
```diff
@@ -119,33 +127,35 @@ async function targetFACCheck(tokenAddress, amount) {
...
+    const amountsOut = amounts[1];
 
-      if (isTokenFoundryAsset == false && Number(amountsOut) > 0) {
-        return true;
-      } else {
-        return false;
-      }
+    if (isTokenFoundryAsset == false && Number(amountsOut) > 0) {
+      return true;
+    } else {
```
Comment (jpine3528) [inline scripts/Ionic.js:156]: I think else { could be removed and return false directly for code optimization.

File: scripts/Refinery.js [hunk 4]
```diff
@@ -120,23 +121,25 @@ async function targetFACCheck(tokenAddress, amount) {
...
-  if (isTargetTokenFoundryAsset == true && Number(targetFoundryAssetLiquidity) > Number(amount)) {
-      return true;
-  }
-  else{
+  if (
+    isTargetTokenFoundryAsset == true &&
+    Number(targetFoundryAssetLiquidity) > Number(amount)
+  ) {
+    return true;
+  } else {
```
Comment (jpine3528) [inline scripts/Refinery.js:134]: I think else { could be removed and return false directly for code optimization.

File: scripts/Refinery.js [hunk 5]
```diff
@@ -160,120 +162,146 @@ async function isTargetRefineryCheck(tokenAddress, amount) {
...
-    return true; 
-  }else {
-      return false;
-    }
+  if (
+    isTokenFoundryAsset == false &&
+    Number(targetFundMangerLiquidity) > Number(amount)
+  ) {
+    return true;
+  } else {
```
Comment (jpine3528) [inline scripts/Refinery.js:170]: I think else { could be removed and return false directly for code optimization.

**Review by taha-abbasi — APPROVED** [review]

**Pull request merged**